compute_tools/src/bin/compute_ctl.rs

```diff
@@ -276,14 +270,53 @@ fn main() -> Result<()> {
         }
     };
 
-    // start the vm-monitor if directed to
-    let vm_monitor = vm_monitor_addr.map(|addr| {
-        tokio::spawn(vm_monitor::start(Box::leak(Box::new(vm_monitor::Args {
-            cgroup: cgroup.cloned(),
-            pgconnstr: file_cache_connstr.cloned(),
-            addr: addr.clone(),
-        }))))
-    });
+    // Start the vm-monitor if directed to. The vm-monitor only runs on linux
+    // because it requires cgroups.
+    cfg_if::cfg_if! {
+        if #[cfg(target_os = "linux")] {
```

Comment:
is it worth using `cfg_if!` here instead of just
```rs
#[cfg(target_os = "linux")]
{
    // your code here
};
```

?

(or, does rustfmt clean up the semicolon?)

Reply:
I like using `cfg_if` because it creates all of the variables in the outer scope. That way we don't have to do something like:
```rust
#[cfg(target_os = "linux")]
let (x, y, z) = {
    ...
    (x, y, z)
};
```

Another nice benefit in particular is that we create the runtime in the outer scope, so there's not confusion over whether it gets dropped when it goes out of scope (it doesn't).

Reply:
Ohhh interesting. TIL!